src/v/kafka/server/replicated_partition.cc

```diff
@@ -370,8 +384,8 @@ replicated_partition::get_leader_epoch_last_offset_unbounded(
 
     // Check cloud storage for a viable offset.
     if (
-      _partition->is_remote_fetch_enabled()
-      && _partition->cloud_data_available()) {
+      is_read_replica
+      || (_partition->is_remote_fetch_enabled() && _partition->cloud_data_available())) {
```

Comment:
I think in the case `is_read_replica && !cloud_data_available()`, we should explicitly return nullopt. At least it seems like if the cloud partition is empty, we could hit some asserts:
- https://github.com/redpanda-data/redpanda/blob/39062df760c7ce9cfcc8770c2247c969593cb623/src/v/cloud_storage/async_manifest_view.cc#L696
- https://github.com/redpanda-data/redpanda/blob/39062df760c7ce9cfcc8770c2247c969593cb623/src/v/cluster/partition.cc#L315

Reply:
In test if the request is performed before we sync it returns -1, but I added the fix just in case